export_saved.py

```diff
@@ -86,105 +87,179 @@ def get_args(argv):
                         action="store_true")
     parser.add_argument("-up", "--upvoted", help="get upvoted posts instead of saved posts",
                         action="store_true")
+    parser.add_argument("-all", "--all", help="get upvoted, saved, comments and submissions",
+                        action="store_true")
 
     args = parser.parse_args(argv)
     return args
 
 
-def login(args):
-    """login.
-
-    Args:
-        args (argparse.Namespace): Parsed arguments.
-
-    Returns:
-        Reddit object instance.
-    """
-    username = None
-    password = None
-    client_id = None
-    client_secret = None
-    if args.username and args.password and args.client_id and args.client_secret:
-        username = args.username
-        password = args.password
-        client_id = args.client_id
-        client_secret = args.client_secret
-    else:
-        try:
-            import AccountDetails
-            username = AccountDetails.REDDIT_USERNAME
-            password = AccountDetails.REDDIT_PASSWORD
-            client_id = AccountDetails.CLIENT_ID
-            client_secret = AccountDetails.CLIENT_SECRET
-        except ImportError:
+class Reddit():
```

Comment:
The whole read write csv and convert it to bookmark is legacy code. No request change or and it is working, so that is why it is keep there. 

And for class vs function I will answer below. Not on this review,